src/main/java/com/wavefront/sdk/direct_ingestion/WavefrontDirectIngestionClient.java

```diff
@@ -34,32 +39,93 @@
  *
  * @author Vikram Raman (vikram@wavefront.com)
  */
-public class WavefrontDirectIngestionClient extends AbstractDirectConnectionHandler
-    implements WavefrontMetricSender, WavefrontHistogramSender, WavefrontTracingSpanSender {
+public class WavefrontDirectIngestionClient implements WavefrontMetricSender,
+    WavefrontHistogramSender, WavefrontTracingSpanSender, BufferFlusher, Runnable, Closeable {
 
   private static final String DEFAULT_SOURCE = "wavefrontDirectSender";
   private static final Logger LOGGER = Logger.getLogger(
       WavefrontDirectIngestionClient.class.getCanonicalName());
-  private static final int MAX_QUEUE_SIZE = 50000;
-  private static final int BATCH_SIZE = 10000;
-
-  private final LinkedBlockingQueue<String> metricsBuffer =
-      new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
-  private final LinkedBlockingQueue<String> histogramsBuffer =
-      new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
-  private final LinkedBlockingQueue<String> tracingSpansBuffer =
-      new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
 
   private final AtomicInteger failures = new AtomicInteger();
+  private final int batchSize;
+  private final LinkedBlockingQueue<String> metricsBuffer;
+  private final LinkedBlockingQueue<String> histogramsBuffer;
+  private final LinkedBlockingQueue<String> tracingSpansBuffer;
+  private final DataIngesterAPI directService;
+  private final ScheduledExecutorService scheduler;
+
+  public static class Builder {
+    // Required parameters
+    private final String server;
+    private final String token;
+
+    // Optional parameters
+    private int maxQueueSize = 50000;
+    private int batchSize = 10000;
+    private int flushIntervalSeconds = 1;
```

Comment:
Make this milliseconds? We can default to `1000ms`, but that would allow support for sub second flush granularity.